include/openmc/urr.h

```diff
@@ -16,16 +17,46 @@ namespace openmc {
 
 class UrrData {
 public:
+  // Since we access all of these at once, we want
+  // them contiguous in memory.
+  struct XSSet {
+    double total;
+    double elastic;
+    double fission;
+    double n_gamma;
+    double heating;
+  };
+
   Interpolation interp_;          //!< interpolation type
   int inelastic_flag_;            //!< inelastic competition flag
   int absorption_flag_;           //!< other absorption flag
   bool multiply_smooth_;          //!< multiply by smooth cross section?
-  int n_energy_;                  //!< number of energy points
-  xt::xtensor<double, 1> energy_; //!< incident energies
-  xt::xtensor<double, 3> prob_;   //!< Actual probability tables
+
+  vector<double> energy_; //!< incident energies
+  auto n_energy() const { return energy_.size(); }
+
+  /* The row indexes correspond to the incident energy table, and column
+   * indices correspond to values of the CDF at that energy. For the CDF matrix
+   * below, obviously, values of the CDF are stored. For the xs_values
+   * variable, the columns line up with the index of cdf_values.
+   */
+  xt::xtensor<double, 2> cdf_values_; // Note: must be row major!
+  xt::xtensor<XSSet, 2> xs_values_;
```

Comment:
No. I really am not a fan of having to define enum classes, and then using them to index into stuff. It's a lot cleaner to put in a struct IMO. It makes the intention of the code more clear, and that this dimension of the tensor is always fixed. What tells you more: that this is a 2-tensor of sets of cross sections, or that this is a 3-tensor of double precision floats?

Your proposed approach would definitely work equivalently, but since we have to read in a temporary xtensor anyway, I think it makes sense to go ahead and use the struct. And we'd still have to pull out the CDF values either way.